### ISSUE TITLE
Compute v2: access_ip_v4 attribute is not correct

### ISSUE THREAD

**Issue by jeka** [terraform-providers/terraform-provider-openstack#699]
Hi there,
I am not sure if this a documentation issue or bug.
Official documentation about OpenStack provider states:

**openstack_compute_instance_v2, exported attribute:
access_ip_v4 - The first detected Fixed IPv4 address or the Floating IP.**

In reality, I don't see this attribute holding a floating IP, only internal IP.
I do remember a time (i think) that there was a way to assign a floating IP directly with openstack_compute_instance resource and maybe documentation reflects this case.
So I wonder how it is supposed to work when using  openstack_compute_floatingip_associate_v2 resource.

```
$ terraform state show module.instance.openstack_compute_instance_v2.this
id                                   = 568c74e6-8dce-4003-8b5d-5f4126bbf387
access_ip_v4                         = 192.168.0.6
access_ip_v6                         = 
all_metadata.%                       = 0
...
network.#                            = 1
network.0.access_network             = false
network.0.fixed_ip_v4                = 192.168.0.6
network.0.fixed_ip_v6                = 
network.0.floating_ip                = 
...

$ terraform state show module.instance.openstack_compute_floatingip_associate_v2.fip
id          = 10.164.132.252/568c74e6-8dce-4003-8b5d-5f4126bbf387/192.168.0.6
fixed_ip    = 192.168.0.6
floating_ip = 10.164.132.252
instance_id = 568c74e6-8dce-4003-8b5d-5f4126bbf387
region      = RegionOne
```

My problem is that ansible's dynamic inventory generator (for example https://github.com/mantl/terraform.py) based of tfstate file is not working as expected.

Terraform v0.11.11
provider.openstack v1.16.0

**Comment by jtopjian** [comment]
@jeka I apologize for the delay.

> I do remember a time (i think) that there was a way to assign a floating IP directly with openstack_compute_instance

Correct. This was possible a long time ago. It was convenient, but was quite difficult to manage. As a result, this functionality was removed. This was a breaking change and it was done shortly after Terraform split the providers into separate repositories and this repo was at version v0.1.0: https://github.com/terraform-providers/terraform-provider-openstack/commit/3d426884761ca431da5ed361bf59f9c4a26e1d90

You are correct that one part of the docs was not updated to reflect this and I apologize about that. I'll get this fixed.

> My problem is that ansible's dynamic inventory generator (for example https://github.com/mantl/terraform.py) based of tfstate file is not working as expected.

This change was definitely a breaking change and would affect projects such as Mantl. Again, I apologize -- breaking changes are not fun. I recommend that projects such as Mantl update their code to read the `openstack_compute_floatingip_associate_v2` resource instead.

An alternative solution is to use something like https://github.com/nbering/terraform-provider-ansible. This allows you to create Ansible inventory resources which provides greater flexibility with Ansible's inventory.

In order to use this type of system, you'll need to use two pieces:

1. The Terraform provider linked above
2. An Ansible dynamic inventory script. nbering has a python-based one here: https://github.com/nbering/terraform-inventory/ and I have a Go-based one here: https://github.com/jtopjian/ansible-terraform-inventory

Please let me know if you have any questions.

**Comment by jeka** [comment]
@jtopjian Thank you for the explanation and pointers.

Although in my tests I didn't manage to get your Go-based inventory generator working for some reason. nbering's python version works, even with Python 3.

**Comment by jtopjian** [comment]
@jeka I'm glad to hear you've found a working solution :)

I'd be happy to dig into why my inventory script isn't working, but only if you had the time. It's something I use daily, so I know it works, but maybe you've run into a bug.

I've merged #703, so I'm going to close this issue. But please let me know if you have any other questions.
